src/server/utilities.ts

```diff
@@ -292,4 +292,54 @@ namespace ts.server {
             deleted(oldItems[oldIndex++]);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * clears already present map (!== undefined) by calling onDeleteExistingValue callback before deleting that key/value

```

Comment:
Could this just call `map.clear()` at the end instead of calling `delete()` incrementally?
Nit: I don't think it's so important to call out that the parameter needs to `!== undefined`.